scriptlets/refresh_zapper_if_needed

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Update the Zapper snap (and firmware) if needed
+#
+# Description:
+#
+# If a Zapper is part of the setup, refresh the snap
+# to the target channel (default is `beta`), upgrade firmware
+# and wait until the service is stable. 
+
+usage() {
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) [--channel]"
+}
+
+check_for_zapper () {
+    (
+        trap EXIT
+        source "$(dirname ${BASH_SOURCE[0]})/defs/check_for_zapper_ip" > /dev/null
+    )
+}
+
+wait_zapper_service_stable() {
```

Comment:
Perhaps it would be worth adding a comment here so that anyone reading this doesn't have to spend any time understanding the logic. Something like "Wait until the number of zapper addons is stable for max_stable_checks consecutive times"

Reply:
it's now described in its own "docstring"